### ISSUE TITLE
spark read csv/json would get empty/unbalanced partitions?

### ISSUE THREAD

**Issue by hkvision** [intel-analytics/analytics-zoo#2467]
Currently we specify numSlices or directly call rdd.repartition, would there be chances that it gives empty partitions or unbalanced partitions?

Previously there is, since the number of partitions is determined by the total number of cores. If the number of files is small, then there is chances to get empty or unbalanced partitions.

For example, if I input a folder with 2 files and then I repartition to 2, one partition gets 2 files and the other is empty.
If I input a folder with 8 files and then repartition to 2, one partition gets 6 files and the other gets 2.

After https://github.com/intel-analytics/analytics-zoo/pull/2466 the above two cases seem not happen. Would there be other possibilities to get empty or unbalanced partitions?

cc @jason-dai

**Comment by jenniew** [comment]
If num of files is less than num_workers, there may get empty partition, can runner handle this?

**Comment by hkvision** [comment]
> f files is less than num

No, currently we assume the number of workers are smaller or equal to partitions. Otherwise a worker has no data to train and there is no need to create so many workers.

We can add an assert in MXNet Estimator.
